### DISCUSSION TITLE
Problems the check that the module boptestGymEnv

### DISCUSSION THREAD

**Discussion by kirachid** [ibpsa/project1-boptest#648] [Q&A - Software] [answered]
good evening Ettore I hope you are well please help me i'm interested in the BOPTEST/project1 project especially bestest_hydronic_heat_pump i have a problem to check that the module boptestGymEnv is correctly installed in your Python environment

**Comment by kirachid** [comment]
Code execution problem

import sys
sys.path.insert(0,'boptestGymService')
from boptestGymService.boptestGymEnv import BoptestGymEnv

# url for the BOPTEST service
url = 'https://api.boptest.net'

# Instantiate environment
env = BoptestGymEnv(url                   = url,
                    testcase              = 'bestest_hydronic_heat_pump',
                    actions               = ['oveHeaPumY_u'],
                    observations          = {'reaTZon_y':(280.,310.)},
                    random_start_time     = False,
                    start_time            = 31*24*3600,
                    max_episode_length    = 24*3600,
                    warmup_period         = 24*3600,
                    step_period           = 3600)

**Comment by EttoreZ** [comment]
Hey Rachid, as a I mentioned I actually never used the Boptest Gym envinroment. Did you follow the guide reported here [BoptestGymQuickStart](https://github.com/ibpsa/project1-boptest-gym) in the quick start section? If you tried and it does not work, can you post the script you are using and the errors you are getting? With the information you provided we cannot help you.

**Reply by kirachid** [reply]
Hey Ettoer, these script I'm using and the errors I'm getting

import sys

# Assurez-vous que le répertoire contenant boptestGymService est dans le chemin de recherche
sys.path.insert(0, 'boptestGymService')

from boptestGymService.boptestGymEnv import BoptestGymEnv

# URL pour le service BOPTEST
url = 'https://api.boptest.net'

try:
    # Instancier l'environnement
    env = BoptestGymEnv(url=url,
                        testcase='bestest_hydronic_heat_pump',
                        actions=['oveHeaPumY_u'],
                        observations={'reaTZon_y': (280., 310.)},
                        random_start_time=False,
                        start_time=31 * 24 * 3600,
                        max_episode_length=24 * 3600,
                        warmup_period=24 * 3600,
                        step_period=3600)
except Exception as e:
    print("Erreur lors de l'instanciation de l'environnement:", e)
    sys.exit(1)



the errors I'm getting

Traceback (most recent call last):
  File "c:\Users\Rchid\PycharmProjects\SALIM\BOPTEST execute", line 317, in <module>
    from boptestGymService.boptestGymEnv import BoptestGymEnv
ModuleNotFoundError: No module named 'boptestGymService.boptestGymEnv'

**Reply by EttoreZ** [reply, marked as answer]
Have you followed the instructions from [BoptestGymQuickStart](https://github.com/ibpsa/project1-boptest-gym):

1. "Create a conda environment from the environment.yml file provided (instructions [here](https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#creating-an-environment-from-an-environment-yml-file)).
2. Check out the boptest-gym-service branch and run the example below that uses the [Bestest hydronic case with a heat-pump](https://github.com/ibpsa/project1-boptest/tree/master/testcases/bestest_hydronic_heat_pump) and the [DQN algorithm](https://stable-baselines3.readthedocs.io/en/master/modules/dqn.html) from Stable-Baselines:"
It seems to me you did not install the BOPTESTgym correctly or you are running the script from the wrong folder.
